src/components/pages/Chat/RoomNameLoader.tsx

```diff
@@ -20,13 +21,22 @@ export default function RoomNameLoader({ roomId }: Props) {
             }
 
             const stream = await streamrClient.getStream(roomId)
-
-            dispatch({
-                type: ActionType.RenameRoom,
-                payload: {
-                    [roomId]: stream.description || '',
-                },
-            })
+            try {
+                const description = getRoomMetadata(stream.description)
+                dispatch({
+                    type: ActionType.RenameRoom,
+                    payload: {
+                        [roomId]: description.name || '',
+                    },
+                })
+            } catch (e) {
+                dispatch({
+                    type: ActionType.RenameRoom,
+                    payload: {
+                        [roomId]: stream.id || '',
+                    },
+                })
+            }
```

Comment:
For a better overview of what exactly is the vulnerable part of it (thing we do the `try…catch` for) we could:

```js
let roomName = stream.id

try {
    roomName = getRoomMetadata(stream.description).name
} catch (e) {}

dispatch({
    type: ActionType.RenameRoom,
    payload: {
        [roomId]: roomName || '',
    },
})
```

Reply:
`try…catch` and `throw` are, to me, one of the most underused parts of js. I can't get enough of it. :)